.github/workflows/cleanup-az-resources-wip.yml

```diff
@@ -0,0 +1,92 @@
+on:
+  pull_request:
+    branches: [users/v-shivakumar/AB#86411]
+    types: [closed]
+
+permissions:
+   id-token: write
+
+name: FullCleanupAzureResources
+
+jobs:
+  cleanup:
+    runs-on: ubuntu-latest
+    steps:
+  
+      - name: Login via Az module
+        uses: azure/login@v1.4.1
+        with:
+          client-id: ${{ secrets.AZ_DEVOPS_APP_ID }}
+          tenant-id: ${{ secrets.AZ_DEVOPS_TENANT_ID }}
+          subscription-id: ${{ secrets.AZ_DEVOPS_SUBSCRIPTION_ID }}
+          enable-AzPSSession: true
+
+      - name: Clean Up Azure Resources
+        uses: azure/powershell@v1.1.0
+        with:
+          azPSVersion: '3.1.0'
+          inlineScript: | 
+            [string]$PRNumber = "${{github.event.pull_request.number}}"
+            $groups = Get-AzResourceGroup -Name "dcm-pr$PRNumber-*-*"  
+            if ($groups -eq $null -or $groups.Count -eq 0)
+            {
+              Write-Host "No Resource groups related to PR $PRNumber are found."
+            }
+            else
+            {
+              Write-Host "$($groups.Count) Resource groups related to PR $PRNumber are found."
+            
+              foreach($group in $groups)
+              {
+                Write-Host "$($groups.Count) Resource groups related to PR $PRNumber are found."
+                [string]$name = $group.ResourceGroupName
+                $kv = Get-AzKeyVault -VaultName $(deploymentName) -ResourceGroupName $group.ResourceGroupName -InRemovedState
```

Comment:
Not sure if we need to purge key vaults here. Once deleted they will be automatically purged in 30 days.

Reply:
separated it as a step. we may want to import pr-variables.yml into this script. But before doing that, I want to experiment this action and check on the outcome.